components/service-operator/apis/database/v1beta1/postgres_types.go

```diff
@@ -279,6 +281,54 @@ func (p *Postgres) GetServiceEntrySpecs(outputs cloudformation.Outputs) ([]map[s
 	return specs, nil
 }
 
+// GetStackPolicy implements cloudformation.Stack to return a serialised form of the stack policy, or nil if one is
+// not needed
+func (p *Postgres) GetStackPolicy() (string, error) {
+	statements := []aws.StatementEntry{
+		{
+			Effect:    "Deny",
+			Action:    []string{"Update:Replace", "Update:Delete"},
+			Principal: "*",
+			Resource:  "LogicalResourceId/RDSCluster",
+		},
+		{
+			Effect:    "Allow",
+			Action:    []string{"Update:Modify"},
+			Principal: "*",
+			Resource:  "LogicalResourceId/RDSCluster",
+		},
+	}
+	instanceCount := p.Spec.AWS.InstanceCount
+	if instanceCount < 1 {
+		instanceCount = DefaultInstanceCount
+	}
+
+	for i := 0; i < instanceCount; i++ {
+		statements = append(statements, aws.StatementEntry{
+			Effect:    "Deny",
+			Action:    []string{"Update:Replace", "Update:Delete"},
+			Principal: "*",
+			Resource:  fmt.Sprintf("LogicalResourceId/RDSDBInstance%d", i),
+		})
+		statements = append(statements, aws.StatementEntry{
+			Effect:    "Allow",
+			Action:    []string{"Update:Modify"},
+			Principal: "*",
+			Resource:  fmt.Sprintf("LogicalResourceId/RDSDBInstance%d", i),
+		})
+	}
+
+	stackPolicy := aws.StackPolicyDocument{
```

Comment:
it feels like we should just return here and leave the caller to deal with json marshalling etc.

Reply:
Done.